### DISCUSSION TITLE
How to change parameters order in iter_commits

### DISCUSSION THREAD

**Discussion by ixSci** [gitpython-developers/GitPython#1980] [Q&A] [answered]
Hello, I have the following simple line: `commits = repo.iter_commits(reference, **{'not': True}, all=True)` which generates the following: `git.cmd:Popen(['git', 'rev-list', '--not', '--all', 'REF'...`. Unfortunately this order is wrong and produces no commits as the command should look like this: `git rev-list REF --not --all`.

Is there a way I can achieve this with GitPython?

**Comment by Byron** [comment, marked as answer]
I think this section on [using Git directly](https://gitpython.readthedocs.io/en/stable/tutorial.html#using-git-directly) would have the answers and what's theoretically possible.

Here, however, I since he method only provides `kvargs`, the order is probably not defined anymore. The only way to control it is to use Git directly, i.e. `repo.git.rev_list('--not', '--all')`, and to parse the result by hand.
